pkg/monitortestlibrary/pathologicaleventlibrary/duplicated_event_patterns.go

```diff
@@ -2,419 +2,545 @@ package pathologicaleventlibrary
+	// never hooked up and was just ignored everywhere. We do not have the capability to detect if
+	// events were within specific test suites yet. Leaving them as an always allow for now.
 	{
-		Regexp: regexp.MustCompile(`reason/ErrorReconcilingNode.*annotation not found for node.*macAddress annotation not found for node`),
-		BZ:     "https://issues.redhat.com/browse/OCPBUGS-13400",
+		Name: "ScaledReplicaSet",
+		LocatorKeyRegexes: map[monitorapi.LocatorKey]*regexp.Regexp{
+			monitorapi.LocatorNamespaceKey:  regexp.MustCompile(`(openshift-controller-manager|openshift-route-controller-manager)`),
+			monitorapi.LocatorDeploymentKey: regexp.MustCompile(`(controller-manager|route-controller-manager)`),
+		},
+		MessageReasonRegex: regexp.MustCompile(`^ScalingReplicaSet$`),
+		MessageHumanRegex:  regexp.MustCompile(`\(combined from similar events\): Scaled (down|up) replica set.*controller-manager-[a-z0-9-]+ to [0-9]+`),
 	},
+
+	AllowBackOffRestartingFailedContainer,
+
+	AllowOVNReadiness,
+	AllowImagePullFromRedHatRegistry,
+	EtcdRequiredResourcesMissing,
+	EtcdClusterOperatorStatusChanged,
+	ProbeErrorTimeoutAwaitingHeaders,
+	ProbeErrorLiveness,
+	ReadinessFailed,
+	ProbeErrorConnectionRefused,
+	NodeHasNoDiskPressure,
+	NodeHasSufficientMemory,
+	NodeHasSufficientPID,
+	FailedScheduling,
+	ErrorUpdatingEndpointSlices,
+	MarketplaceStartupProbeFailure,
 }
 
-// IsRepeatedEventOKFunc takes a monitorEvent as input and returns true if the repeated event is OK.
-// This commonly happens for known bugs and for cases where events are repeated intentionally by tests.
-// Use this to handle cases where, "if X is true, then the repeated event is ok".
-type IsRepeatedEventOKFunc func(monitorEvent monitorapi.Interval, kubeClientConfig *rest.Config, times int) (bool, error)
+// Some broken out matchers are re-used in a test for that specific event
 
-var AllowedRepeatedEventFns = []IsRepeatedEventOKFunc{
-	isConsoleReadinessDuringInstallation,
-	isConfigOperatorReadinessFailed,
-	isConfigOperatorProbeErrorReadinessFailed,
-	isConfigOperatorProbeErrorLivenessFailed,
-	isOauthApiserverProbeErrorReadinessFailed,
-	isOauthApiserverProbeErrorLivenessFailed,
-	isOauthApiserverProbeErrorConnectionRefusedFailed,
-	isErrorReconcilingNode,
-	isFailedScheduling,
-	isOperatorStatusChanged,
+// reason/(?P<REASON>Unhealthy) (?P<MSG>Readiness probe failed:.*$)`
+var AllowOVNReadiness = &PathologicalEventMatcher{
+	Name: "OVNReadinessProbeFailed",
+	LocatorKeyRegexes: map[monitorapi.LocatorKey]*regexp.Regexp{
+		monitorapi.LocatorNamespaceKey: regexp.MustCompile(`openshift-ovn-kubernetes`),
+		monitorapi.LocatorPodKey:       regexp.MustCompile(`ovnkube-node-`),
+	},
+	MessageReasonRegex: regexp.MustCompile(`^Unhealthy$`),
+	MessageHumanRegex:  regexp.MustCompile(`Readiness probe failed:`),
 }
 
-var AllowedSingleNodeRepeatedEventFns = []IsRepeatedEventOKFunc{
-	isConnectionRefusedOnSingleNode,
```

Comment:
Pretty sure this was not active, the events were already blanket allowed by: regexp.MustCompile("Unhealthy Readiness probe failed"),